src/Raven.Studio/typescript/common/extensions.ts

```diff
@@ -191,9 +193,140 @@ class extensions {
+        ko.bindingHandlers["requiredAccess"] = {
+            init: (element: Element,
+                   valueAccessor: KnockoutObservable<Raven.Client.ServerWide.Operations.Certificates.DatabaseAccess>,
+                   allBindings) => {
+                
+                const requiredAccessLevel = ko.unwrap(valueAccessor());
+                const levelTypes = ["Read", "ReadWrite", "Admin"];
+
+                if (!_.includes(levelTypes, requiredAccessLevel)) {
+                    throw new Error(`Invalid requiredAccess level. Value provided: ${requiredAccessLevel}. Possible values are: ${levelTypes}.`);
+                }
+                const bindings = allBindings();
+                const bindingsArray = Object.keys(bindings);
+                const requiredAccessBindingLocation = bindingsArray.indexOf("requiredAccess");
+                
+                const activeDatabase = activeDatabaseTracker.default.database();
+                if (activeDatabase) {
+                    if (bindings.visible) {
+                        this.verifyBindingLocation(bindingsArray, "visible", requiredAccessBindingLocation);
+                    }
+                    if (bindings.hidden) {
+                        this.verifyBindingLocation(bindingsArray, "hidden", requiredAccessBindingLocation);
+                    }
+                    if (bindings.disable) {
+                        this.verifyBindingLocation(bindingsArray, "disable", requiredAccessBindingLocation);
+                    }
+                    if (bindings.enable) {
+                        this.verifyBindingLocation(bindingsArray, "enable", requiredAccessBindingLocation);
+                    }
+                } else {
+                    throw new Error("Cannot use the 'requiredAccess' binding - no database is active.");
+                }
+
+                const strategyTypes = ["hide", "disable", "visibilityHidden"];
+                
+                if (bindings.requiredAccessOptions) {
+                    const strategy = bindings.requiredAccessOptions.strategy;
+                    if (!_.includes(strategyTypes, strategy)) {
+                        throw new Error(`Invalid requiredAccess strategy. Value provided: ${strategy}. Possible values are: ${strategyTypes}`);
+                    }
 
+                    if (strategy === 'disable') {
+                        const hasDisabledClass = $(element).hasClass("disabled");
+                        if (hasDisabledClass) {
+                            throw new Error("Do not use the 'disabled' class. Please use the disabled/enabled bindings.");
+                        }
+
+                        if (element.tagName === "A") {
+                            throw new Error("Please do not use the 'requiredAccessOptions disable strategy' binding with element of type 'a'.");
+                        }
+                        
+                        const hasDisabledBinding = _.includes(bindingsArray, "disable");
```

Comment:
you can check that directly in allBinidngs:
if (bindings.enable && bindings.disable) {
   // throw
}

Reply:
done